repl/scala-2.11/src/main/scala/com/cloudera/livy/repl/SparkInterpreter.scala

```diff
@@ -0,0 +1,183 @@
+
+    sparkILoop = new SparkILoop(None, new JPrintWriter(outputStream, true))
+    sparkILoop.settings = settings
+    sparkILoop.createInterpreter()
+    sparkILoop.initializeSynchronous()
+
+    restoreContextClassLoader {
+      sparkILoop.setContextClassLoader()
+
+      var classLoader = Thread.currentThread().getContextClassLoader
+      while (classLoader != null) {
+        if (classLoader.getClass.getCanonicalName ==
+          "org.apache.spark.util.MutableURLClassLoader") {
+          val extraJarPath = classLoader.asInstanceOf[URLClassLoader].getURLs()
+            // Check if the file exists. Otherwise an exception will be thrown.
+            .filter { u => u.getProtocol == "file" && new File(u.getPath).isFile }
+            // Livy rsc and repl are also in the extra jars list. Filter them out.
+            .filterNot { u => Paths.get(u.toURI).getFileName.toString.startsWith("livy-") }
+
+          extraJarPath.foreach { p => debug(s"Adding $p to Scala interpreter's class path...") }
+          sparkILoop.addUrlsToClassPath(extraJarPath: _*)
+          classLoader = null
+        } else {
+          classLoader = classLoader.getParent
+        }
+      }
+
+      createSparkContext(conf)
+    }
+
+    sparkContext
+  }
+
+  override def close(): Unit = synchronized {
+    if (sparkHttpServer != null) {
+      val method = sparkHttpServer.getClass.getMethod("stop")
+      method.setAccessible(true)
+      method.invoke(sparkHttpServer)
+      sparkHttpServer = null
+    }
+    if (sparkContext != null) {
+      sparkContext.stop()
+      sparkContext = null
+    }
+
+    if (sparkILoop != null) {
+      sparkILoop.closeInterpreter()
+      sparkILoop = null
+    }
+  }
+
+  override protected def isStarted(): Boolean = {
+    sparkContext != null && sparkILoop != null
+  }
+
+  override protected def interpret(code: String): Result = {
+    sparkILoop.interpret(code)
+  }
+
+  override protected def valueOfTerm(name: String): Option[Any] = {
+    // IMain#valueOfTerm will always return None, so use other way instead.
+    Option(sparkILoop.lastRequest.lineRep.call("$result"))
+  }
+
+  protected def bind(name: String, tpe: String, value: Object, modifier: List[String]): Unit = {
+    sparkILoop.beQuietDuring {
+      sparkILoop.bind(name, tpe, value, modifier)
+    }
+  }
+
+  private def createTempDir(rootDir: String): File = {
+    try {
+      val method = Class.forName("org.apache.spark.util.Util")
+        .getMethod("createTempDir", classOf[String], classOf[String])
+      method.setAccessible(true)
+      method.invoke(null, rootDir, "spark").asInstanceOf[File]
+    } catch {
+      case NonFatal(e) =>
+        val dir = new File(rootDir, s"spark-${UUID.randomUUID().toString}")
+        dir.mkdir()
+        dir
+    }
+  }
+
+  private def startHttpServer(outputDir: File): Option[(Object, String)] = {
+    try {
+      val httpServerClass = Class.forName("org.apche.spark.HttpServer")
```

Comment:
You have a typo there.

Also, isn't SparkEnv responsible for starting this? Why does it have to be done manually in the 2.11 repl? (See how the 2.10 repl doesn't do this.)